### ISSUE TITLE
Is there no "Reverse Path" method???

### ISSUE THREAD

**Issue by SmartArtsStudio** [adobe-webplatform/Snap.svg#443]
Is there no "Reverse Path" method?
...To reverse the 'd' attribute's co-ordinate string, from undesirable svg editor outputs. Or is there a common WORKING technique something along the lines of:
```javascript
var pathNode= svg.getElementById( 'pathNode' );
// get 'd' value string
var pathString = testPath.attributes.d.nodeValue;
// convert string to absolute coordinates
var absolutePath = Snap.path.toAbsolute( Snap.parsePathString( pathString ) );
// reverse coordinates
var reversedPath = absolutePath.reverse();
// assign reversed coordinate string
svg.getElementById( 'pathNode' ).setAttribute( 'd', absolutePath.toString() );
```
Obviously this fails because it reverses the coordinates but does not move the drawing notation.

I am just really surprised that this is not already built into the library?
(I know paths can be reversed manually in an svg editor, I looking for a method to handle output paths I do not control.)

**Comment by ibrierley** [comment]
Not aware of a way myself, whats the issue ? Just wondering if there's an workaround.

**Comment by SmartArtsStudio** [comment]
I am creating an SVG "draw on effect" path animation renderer, which animates strokes over time from the start of their path to the end. So for the case where complex paths were constructed in the opposite direction to the way they needs to animate... 
I thought a "reverse path" method solution: to correct the <path> 'd' attributes vertices and drawing commands, would be a much simpler solution than a renderer that identifies when paths are animating in the wrong direction and renders them backwards (the illusion of path reversed) while the others go forwards simultaneously.

**Comment by ibrierley** [comment]
Well if you are using something like the stroke-dasharray/offset method, I guess you can just reverse the offset increments. Not quite sure how you would automate deciding if a path needs reversing visually in either case though.. it may be worth doing a jsfiddle

**Comment by SmartArtsStudio** [comment]
Hello, 
I have an even better solution 80% built, I have solved how to parse a path string and reverse drawing commands and arguments accurately, there are a couple extra convenience methods relating to shape element I would like to add on top, and nodejs compatibility I need to look into. This will possibly only take a day or two to find the spare time to finish and tidy up. 

Happy to contribute the solution to snap.svg  when I have some spare time if snap is interested (which it should be! :), if it is could you share some information about the process for proposing contributions. 

Thanks for looking into this.

**Comment by SmartArtsStudio** [comment]
For anyone else looking for the "Reverse SVG Path" solution. I have published a well documented light weight library called:
smart-svg-path 
here: https://github.com/SmartArtsStudio/smart-svg-path

I highly recommend SmartSVGPath.reverse() and SmartSVGPath.setFirstVertex() which I could not find anywhere. It also has some other neat tricks up its sleeve, I am likely to add more clever tricks eventually as I built smart-svg-path in order to build another project. I am sure that projects development will probably highlight other needed hard to find path manipulation solutions.
